src/midi/midi.cpp

```diff
@@ -227,29 +229,35 @@ class MIDI:public Module_base{
 		midi.status=0x00;
 		midi.cmd_pos=0;
 		midi.cmd_len=0;
-		if (!strcasecmp(dev,"default")) goto getdefault;
+		// Value "default" exists for backwards-compatibility.
+		if (dev == "auto" || dev == "default")
+			goto getdefault;
 		handler=handler_list;
 		while (handler) {
-			if (!strcasecmp(dev,handler->GetName())) {
+			if (dev == handler->GetName()) {
 				if (!handler->Open(conf)) {
-					LOG_MSG("MIDI: Can't open device:%s with config:%s.",dev,conf);
+					LOG_MSG("MIDI: Can't open device: %s with config: '%s'",
+					        dev.c_str(), conf);
 					goto getdefault;
 				}
 				midi.handler=handler;
 				midi.available=true;
-				LOG_MSG("MIDI: Opened device:%s",handler->GetName());
+				LOG_MSG("MIDI: Opened device: %s",
+				        handler->GetName());
 				return;
 			}
 			handler=handler->next;
 		}
-		LOG_MSG("MIDI: Can't find device:%s, finding default handler.",dev);
+		LOG_MSG("MIDI: Can't find device: %s, using default handler.",
+		        dev.c_str());
 getdefault:
 		handler=handler_list;
```

Comment:
goto are weaving a nice logic rug :-)
Suggest adding a TODO to cleanup this logic and eliminate the goto's.